jerry-core/api/jerry.c

```diff
@@ -934,26 +934,42 @@ jerry_get_value_from_error (jerry_value_t value, /**< api value */
 } /* jerry_get_value_from_error */
 
 /**
- * Set the error flag if the value is not an error reference.
+ * Create error from an api value.
+ *
+ * Create error value from an api value. If the second argument is true
+ * it will release the input api value.
+ *
+ * @return api error value
  */
-void
-jerry_value_set_error_flag (jerry_value_t *value_p) /**< api value */
+jerry_value_t
+jerry_create_error_from_value (jerry_value_t value, /**< api value */
+                                      bool release) /**< release api value */
 {
   jerry_assert_api_available ();
+  jerry_value_t ret_value;
 
-  if (JERRY_UNLIKELY (ecma_is_value_error_reference (*value_p)))
+  if (JERRY_UNLIKELY (ecma_is_value_error_reference (value)))
   {
     /* This is a rare case so it is optimized for
      * binary size rather than performance. */
-    if (!jerry_value_is_abort (*value_p))
+    if (!jerry_value_is_abort (value))
     {
-      return;
+      return value;
+    }
+
+    ret_value = jerry_get_value_from_error (value, release);
+  }
+  else
+  {
+    ret_value = value;
+    if (release)
+    {
+      jerry_release_value (value);
```

Comment:
This is not correct as well. You use `value` through `ret_value` after released.